### DISCUSSION TITLE
Passing public properties

### DISCUSSION THREAD

**Discussion by thanosalexander** [livewire/livewire#2178] [1. Help]
I want to use an array of components input,select etc to render a Form livewire component. Of course this array is dynamic and i would like to know if there is any way to init to call form, and pass public properties.
I use rules too and i want to validate
This way i pass data!
```
  <livewire:client.user.registration.registration-web-form
        :components="$components"
        :rules="$rules"
        :form-data="$formData"
   />
```

I tried wrap form data with a key `formData` and then on rendering i use `wire:model='formData.{{ $component['name'] }}'`, so to communicate with component.
This way i can handle updates , but i don't prefer. 
It could be nice if we had the ability to pass the parameters as params in constructor and have them available.

Eg this way i'll use `wire:model='{{ $component['name'] }}`

**Comment by joshhanley** [comment]
@thanosalexander any chance you could try and explain what you're after in some more detail?
I personally am struggling to understand what you're trying to achieve, and as such can't be of much help yet.

Please share as much info and code as you can, that you think will help define the issue.

Thanks!

**Reply by thanosalexander** [reply]
Hi josh. My target was to create a livewire form component, in which i'd pass some form data. 
Theese data of course needs to be validated with livewire's validation features!
And now i return to my issue which is if i could have theese data setted as public properties(like a routine class construction), without having to wrap them and do alll the job(wire:models,validations...) through this.
I hope i helped you!

**Comment by joshhanley** [comment]
@thanosalexander it sounds to me like you want to have a Livewire component which represents your form, in which you could pass it an array of properties, for which you want to dynamically create form inputs for, allow submission, and ensure it all gets validated correctly. Does that sound right?

Have you experimented with anything, if so can you share you code of ideally what you would like to happen, then I might be able to help you get it to work?
